cmd/minishift/cmd/profile/profile_copy.go

```diff
@@ -0,0 +1,132 @@
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/spf13/cobra"
+)
+
+var (
+	profileCopyCmd = &cobra.Command{
+		Use:   "copy SRC_PROFILE_NAME NEW_PROFILE_NAME",
+		Short: "Copy configurations from an existing profile to a new profile.",
+		Long:  "Copy add-on and config configurations from an existing profile to a new profile.",
+		Run:   copyProfile,
+	}
+)
+
+func copyProfile(cmd *cobra.Command, args []string) {
+	validateCopyProfileCmd(args)
+	srcProfile := args[0]
+	newProfile := args[1]
+
+	if !cmdUtil.IsValidProfile(srcProfile) {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Profile %s does not exist", srcProfile))
+	}
+
+	if cmdUtil.IsValidProfile(newProfile) {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Profile '%s' already exists. You must provide a non-existant profile name", newProfile))
+	}
+	copySrcToNewProfile(srcProfile, newProfile)
+
+}
+
+func copySrcToNewProfile(srcProfile, newProfile string) {
+	srcConfFile := constants.GetProfileConfigFile(srcProfile)
+	sourceConfig, err := config.ReadViperConfig(srcConfFile)
+	if err != nil {
+		atexit.ExitWithMessage(1, err.Error())
+	}
+
+	newConfFile := constants.GetProfileConfigFile(newProfile)
+
+	// Check if the config file for the new profile is not empty
+	newConfig, err := config.ReadViperConfig(newConfFile)
+	if err != nil {
+		atexit.ExitWithMessage(1, err.Error())
+	}
+	if len(newConfig) != 0 {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Can not copy configurations from '%s' to '%s' as '%s' is not empty", srcProfile, newProfile, newConfFile))
+	}
+
+	newInstanceDirs := state.GetMinishiftDirsStructure(constants.GetProfileHomeDir(newProfile))
+	cmdUtil.CreateMinishiftDirs(newInstanceDirs)
+	cmdUtil.EnsureConfigFileExists(newConfFile)
+
+	err = config.WriteViperConfig(newConfFile, sourceConfig)
+	if err != nil {
+		atexit.ExitWithMessage(1, err.Error())
+	}
+	srcProfileDirs := cmdState.GetMinishiftDirsStructure(constants.GetProfileHomeDir(srcProfile))
+
+	// Removing the newly created addon directory as filehelper.CopyDir does not expect the directory to be present.
+	err = os.Remove(newInstanceDirs.Addons)
+	if err != nil {
+		atexit.ExitWithMessage(1, err.Error())
+	}
+
+	// Copying add-ons from source profile to the new profile
+	err = filehelper.CopyDir(srcProfileDirs.Addons, newInstanceDirs.Addons)
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Copying add-ons to profile '%s' failed : %s", newProfile, err.Error()))
+	}
+
+	newProfileInstanceConfig, err := config.NewInstanceConfig(minishiftConstants.GetProfileInstanceConfigPath(newProfile))
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating config for profile '%s' : %s", newProfile, err.Error()))
+	}
+
+	srcProfileInstanceConfig, err := config.NewInstanceConfig(minishiftConstants.GetProfileInstanceConfigPath(srcProfile))
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating config for source profile '%s' : %s", srcProfile, err.Error()))
```

Comment:
Why is there a creation needed for the source?

Reply:
@gbraad It is required because we can use any existing profile as source , it does not have to be the active profile e.g. 
```
$ minishift profile list
- minishift	Does Not Exist	(Active)
- test		Does Not Exist

````
```
minishift profile copy test test1
```